app/trip/[id]/components/trip-details-tabs.tsx

```diff
@@ -0,0 +1,105 @@
+import React from 'react'
+import {
+  Box,
+  useColorModeValue,
+  Tabs,
+  TabList,
+  TabPanels,
+  Tab,
+  TabPanel,
+  VStack,
+  Heading
+} from '@chakra-ui/react'
+import { formatDateToDayMonthWeek } from '@/libs/utils'
+import { ActivityCard } from '../components'
+
+export type ActivityType = {
+  id: string
+  timeFrom: string | null | undefined
+  timeTo: string | null | undefined
+  title: string
+  address: string | null | undefined
+}
+
+type TripDetailsTabsProps = {
+  activities: ActivityType[]
+}
+
+export const TripDetailsTabs = ({ activities }: TripDetailsTabsProps) => {
+  const borderColor = useColorModeValue('gray.300', 'gray.600')
+
+  // Sorting by date Time
+  const sortedActivities = activities.sort((a, b) => {
+    const dateA = a.timeFrom
+      ? new Date(a.timeFrom).getTime()
+      : a.timeTo
+      ? new Date(a.timeTo).getTime()
+      : 0
+    const dateB = b.timeFrom
+      ? new Date(b.timeFrom).getTime()
+      : b.timeTo
+      ? new Date(b.timeTo).getTime()
+      : 0
+
+    return dateA - dateB
+  })
+
+  // Grouping by date
+  const activitiesByDate: { [date: string]: ActivityType[] } = {}
+
+  sortedActivities.forEach((activity) => {
+    const date = activity.timeFrom?.split('T')[0]
```

Comment:
If timeFrom is nullable, we can't show activities, even if we have other data like timeTo.
If there's no way, I'd say we better change schema.

Also `T` standing for Timezone depends on DB/GraphQL schema, so I'd say putting this logic into `libs/utils` would be better to hide internal logic.

Reply:
Changed schema and made function for format!
https://github.com/tabi-memo/frontend/pull/17/commits/df484de13db7f1388cf0517e9b1e54d26f70489b